// [[Reason: Function declarations are not fixed]] ////

//// [tests/cases/compiler/declarationEmitFunctionKeywordProp.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,20 +1,31 @@
 
 
 //// [declarationEmitFunctionKeywordProp.d.ts]
 declare function foo(): void;
-declare namespace foo {
-    var _a: boolean;
-    export { _a as null };
-}
 declare function bar(): void;
-declare namespace bar {
-    var async: boolean;
-    var normal: boolean;
-}
 declare function baz(): void;
-declare namespace baz {
-    var _a: boolean;
-    export var normal: boolean;
-    export { _a as class };
-}
-//# sourceMappingURL=declarationEmitFunctionKeywordProp.d.ts.map
\ No newline at end of file
+//# sourceMappingURL=declarationEmitFunctionKeywordProp.d.ts.map
+/// [Errors] ////
+
+declarationEmitFunctionKeywordProp.ts(1,10): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitFunctionKeywordProp.ts(4,10): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+declarationEmitFunctionKeywordProp.ts(8,10): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+
+
+==== declarationEmitFunctionKeywordProp.ts (3 errors) ====
+    function foo(): void {}
+             ~~~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.null = true;
+    
+    function bar(): void {}
+             ~~~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    bar.async = true;
+    bar.normal = false;
+    
+    function baz(): void {}
+             ~~~
+!!! error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    baz.class = true;
+    baz.normal = false;
\ No newline at end of file
